lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -253,6 +258,28 @@ sub _route_path {
   return $path;
 }
 
+sub _security_action {
+  my ($self, $security_settings) = @_;
+  my $security_cb = $self->_security_cb;
+
+  return sub {
+    my $c = shift;
+    my @cc
+      = map { my ($name, $config) = %$_; ($security_cb->{$name}, $config); } @$security_settings;
+    my $wrapper;
+
+    $wrapper = sub {
+      my $c = shift;
+      my $cb = shift @cc || sub { die 'No security handler defined' };
+      $c->$cb(shift @cc, $wrapper);
```

Comment:
I am sure helpers are slick, especially since you use them in their own pseudo-namespace "openapi_security_dummy".
However I find helpers to be very confusing and I remember struggling to understand from where these cryptic $c->$helper_name -things spawn from?

Since you have a specific use-case for the "Security Definitions Object"-implementations it would be more clear to the end-user to find those implementations in
$c->openapi->security->dummy($config, $wrapper);

You can access the App from most everywhere anyway.